### PR TITLE
Add the Google Cloud service and use it to destroy instances (refs #2…

### PR DESCRIPTION
**Pull request by vlasov-evgeny**
…6357)

### CODE REVIEW

File: core/services/gcp_service.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      json_key_io: File.open(@gcp_config['credentials_file']),
+      scope: SCOPE
+    )
+    @service.authorization.fetch_access_token!
+  end
+
+  # Fetch instances list and return instance names.
+  # @return [Array<String>] instance names.
+  def instances_list
+    items = @service.fetch_all do |token|
```
Comment (amvasilyev) [inline core/services/gcp_service.rb]: You do not need items variable here. Just chain the `map` method to the results of calling `fetch_all`.

File: core/services/gcp_service.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  # Returns false if a new vpc resources need to be generated for the current configuration, otherwise true.
+  # @return [Boolean] result.
+  def use_existing_network?
+    @gcp_config['use_existing_network']
+  end
+
+  # Delete instance specified by the it name
+  # @param instance_name [String] name of the instance to delete.
+  def delete_instance(instance_name)
+    return if instance_name.nil?
```
Comment (amvasilyev) [inline core/services/gcp_service.rb]: Can this actually happen? Why do you even pass `nil` to this method?

File: core/commands/destroy_command.rb [hunk 3]
```diff
@@ -113,18 +114,21 @@ def destroy_by_node_name
     vagrant_cleaner = VagrantCleaner.new(@env, @ui)
     vagrant_vm_list = vagrant_cleaner.vm_list
     aws_vm_list = @aws_service.instances_names_list
+    gcp_vm_list = @gcp_service.instances_list
```
Comment (amvasilyev) [inline core/commands/destroy_command.rb:117]: From the code in Session we can see that AWS and GCP are optional. This code will fail if user did not provide AWS or GCP credentials.

File: core/commands/destroy_command.rb [hunk 4]
```diff
@@ -102,29 +102,33 @@ def filter_nodes_by_name(vm_list, node_name)
   def display_all_nodes
     vagrant_cleaner = VagrantCleaner.new(@env, @ui)
     vagrant_vm_list = vagrant_cleaner.vm_list
-    aws_vm_list = @aws_service.instances_names_list
+    aws_vm_list = @aws_service&.instances_names_list || []
```
Comment (amvasilyev) [inline core/commands/destroy_command.rb]: We should discuss: should it be easier to provide `aws_service` all the time?

Comment (vlasov-evgeny) [reply core/commands/destroy_command.rb]: In order not to check for the existence of the service, but in order to return empty data if the service is not configured?

**Pull request merged**